### PR TITLE
Support usage of JupyterLab 2.0.0

### LINKED ISSUE #173: JupyterLab 2.0.0 compatibility
JupyterLab 2.0.0 is out, and this extension have conflicting dependencies with its core packages.

```
Conflicting Dependencies:
JupyterLab              Extension      Package
>=2.0.0 <2.1.0          >=1.0.0 <2.0.0 @jupyterlab/application
>=2.0.0 <2.1.0          >=1.0.0 <2.0.0 @jupyterlab/launcher
```

## References
- [Change log for developers](https://jupyterlab.readthedocs.io/en/stable/getting_started/changelog.html#for-developers)
- [A migration guide](https://jupyterlab.readthedocs.io/en/stable/developer/extension_migration.html#extension-migration)

### PR DESCRIPTION
**Pull request by consideRatio**
Fixes #173

At least, I think this fixes #173, I have not verified the functionality but only verified I could build it without failures after bumping it. Also, note the comment in the second commit for an explanation about the change in tsconfig.json

### CODE REVIEW

**Comment by betatim** [comment]
Do you think we will be able to use this with JupyterLab 1.0 and 2.0 simultaneously or we need separate builds/releases?

**Comment by consideRatio** [comment]
I think simultaneous use is fine as I didn't change the code at all.

File: jupyterlab-server-proxy/tsconfig.json [hunk 1]
```diff
@@ -1,5 +1,6 @@
 {
   "compilerOptions": {
+    "allowSyntheticDefaultImports": true,
```
Comment (betatim) [inline jupyterlab-server-proxy/tsconfig.json]: On purpose or a accidental change?

Comment (consideRatio) [reply jupyterlab-server-proxy/tsconfig.json]: Om purpose, the reason is described in the second commit's commit messages

**Pull request merged**